### PR TITLE
Add BlobFS test resource provider 

### PR DESCRIPTION
**Pull request by shyaler**
Add BlobFS resource provider to support e2e test operations.

### CODE REVIEW

File: cmd/sync.go [hunk 1]
```diff
@@ -262,7 +262,7 @@ func (raw *RawSyncCmdArgs) Cook() (cookedSyncCmdArgs, error) {
 	}
 
 	cooked.followSymlinks = raw.FollowSymlinks
-	if cooked.fromTo != common.EFromTo.LocalFile() && cooked.fromTo != common.EFromTo.LocalBlob() { // Follow symlinks in sync is supported only for local -> file or local -> blob
+	if cooked.followSymlinks && cooked.fromTo != common.EFromTo.LocalFile() && cooked.fromTo != common.EFromTo.LocalBlob() { // Follow symlinks in sync is supported only for local -> file or local -> blob
```
Comment (afekson) [inline cmd/sync.go:265]: Why is this a part of this PR? How does it relates to E2E tests?

Comment (shyaler) [reply cmd/sync.go:265]: its a bug, found when tried to run the e2e tests. don't see a reason to create a separate PR just for this fix.

Comment (afekson) [reply cmd/sync.go:265]: can you please provide more details about the bug?

@OmarKhatib158

**Review by OmarKhatib158 — CHANGES_REQUESTED** [review]

**Review by shmuelberezin — APPROVED** [review]

File: e2etest/declarativeResourceManagers.go [hunk 2]
```diff
@@ -314,6 +320,100 @@ func (r *resourceBlobContainer) createSourceSnapshot(a asserter) {
...
+}
+
+func (r *resourceBlobFSContainer) appendSourcePath(filePath string, useSas bool) {
+	if useSas {
+		r.rawSasURL.Path += "/" + filePath
+	}
+}
+
+func (r *resourceBlobFSContainer) getAllProperties(a asserter) map[string]*objectProperties {
+	panic("Not Implemented")
```
Comment (shmuelberezin) [inline e2etest/declarativeResourceManagers.go:403]: @shyaler, won't it fail in validateProperties?

Comment (OmarKhatib158) [reply e2etest/declarativeResourceManagers.go:403]: I will do it

**Review by OmarKhatib158 — APPROVED** [review]

**Pull request merged**